### PR TITLE
gha: add golangci-lint job, fix warnings; bump API version

### PR DESCRIPTION
**Pull request by kolyshkin**
This
 - reworks the source layout to ensure linters don't choke;
 - slightly improves Makefile and running tests;
 - introduces golangci-lint from Makefile as well as for GitHub Actions;
 - fixes all golangci-lint warnings found;
   - :warning: this includes bumping API to v5 since we changed the prototypes;
 - adds a small `go mod tidy` verify job to GHA;
 - enables some more linters and fixes their warnings (suggested by @rst0git)

See individual commits for more details.

Description for changelog:
```
 - API bumped to v5
 - methods (such as Dump, Restore etc.) now require a pointer to rpc.CriuOpts as the first argument
```

### CODE REVIEW

**Comment by kolyshkin** [comment]
OK, this can't be done until we have a mix of .c and .go files. Trying to fix...

**Comment by kolyshkin** [comment]
@adrianreber @rst0git PTAL. Unfortunately I had to change the public API and thus bump v4 to v5 -- I see no way around it.

**Comment by adrianreber** [comment]
Thanks. Looks good to me. I have no problem doing a v5 based release. @rst0git what do you think about these changes?

File: Makefile [hunk 1]
```diff
@@ -1,55 +1,46 @@
...
-ifeq ($(GOBIN),)
-	GOBIN := $(FIRST_GOPATH)/bin
-endif
 
-all: build test phaul phaul-test
+all: build test phaul-test
 
 lint:
-	@golint -set_exit_status . test phaul
+	golangci-lint run ./...
```
Comment (rst0git) [inline Makefile:7]: What about enabling some of the linters presets?

```
golangci-lint run -p bugs,performance,unused,format ./...
```

Comment (kolyshkin) [reply Makefile:7]: I think this should be addressed separately (and it's better/easier to do so via `.golangci.yml`), once this is merged.

Comment (kolyshkin) [reply Makefile:7]: @rst0git implemented; PTAL

**Comment by kolyshkin** [comment]
(CI for 1.16-rc1 is broken as it was removed; fix in #47)

File: Makefile (hunk 1 above)
Comment (kolyshkin) [reply Makefile:7]: This probably makes the patchset slightly harder to review -- let me know if you want a separate PR.

**Review by rst0git — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (rst0git) [reply Makefile:7]: Thanks!

**Comment by adrianreber** [comment]
@rst0git Do you want to tag the new release? 5.0.0?

**Pull request merged**